### PR TITLE
WebSocket implementation - continuation

### PR DESCRIPTION
**Pull request by KhrulSergey**
1. Define all possible endpoints for Mexc WS
2. Added mechanism to process WS messages through wrap class MexcWSResponse.java and ObjectMapper
3. All Data models that is used in WS should extend WSBaseDto in order to fill additional data from MexcWSResponse wrapper.
4. Added couple Enums and DTO for using in processing WS data.

### CODE REVIEW

File: src/main/java/trade/wayruha/mexc/dto/Order.java [hunk 1]
```diff
@@ -1,33 +1,70 @@
...
+@Getter
+@Setter
+@RequiredArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class Order {
-    private String symbol;
-    private long orderId;
+public class Order extends WSBaseDto {
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/dto/Order.java:20]: This Order dto is not related to web sockets. why do we extend it from WSBaseDto?

File: src/main/java/trade/wayruha/mexc/enums/OrderStatus.java [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+
+@Getter
+@AllArgsConstructor
 public enum OrderStatus {
-    NEW,
-    FILLED,
-    PARTIALLY_FILLED,
-    CANCELED,
-    PARTIALLY_CANCELED
+    NEW (1),
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/enums/OrderStatus.java:11]: is `aliasNumber` used anywhere in the API? I haven't seen that we get a number instead of a string in the response

File: src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java [hunk 3]
```diff
@@ -127,10 +132,14 @@ public void onMessage(WebSocket webSocket, String text) {
...
         try {
-            final T obj = objectMapper.readValue(text, type);
-            callback.onResponse(obj);
+            JavaType typeMap = objectMapper.getTypeFactory().constructParametricType(MexcWSResponse.class, type);
+            MexcWSResponse<T> obj = objectMapper.readValue(text, typeMap);
+            T data = obj.getData();
+            if (nonNull(data) && nonNull(obj.getSymbol())) {
+                data.setSymbol(obj.getSymbol());
+            }
+            callback.onResponse(data);
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java:142]: According to this code, the data model will always have `symbol`. It may be true with this state of API but eventually they will extend it. For example, they will add 'on balance change subscription' which will have nothing to do with trading symbol (e.g., transfers) and our model will be wrong. I vote against having either 1) base model dto (WSBaseDto) OR a `symbol` as a common field for every dto

File: src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java [hunk 4]
```diff
@@ -30,25 +36,137 @@ public WSClientFactory(ApiClient apiClient) {
         this.objectMapper = new ObjectMapper();
     }
 
-    public <T> WebSocketClient<T> orderBookSubscription(Set<String> symbols, int depth, WebSocketCallback<T> callback, Class<T> type) {
+    //<editor-fold desc="Public methods">
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java:39]: why do we have this comments? what for?

File: src/main/java/trade/wayruha/mexc/enums/TopLimit.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package trade.wayruha.mexc.enums;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum TopLimit {
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/enums/TopLimit.java:8]: is it Order Book Depth? should we rename it?

File: src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java [hunk 6]
```diff
@@ -30,25 +36,137 @@ public WSClientFactory(ApiClient apiClient) {
...
-    public <T> WebSocketClient<T> userSpotTradesSubscription(Request connectionRequest, WebSocketCallback<T> callback, Class<T> type) {
-        return new WebSocketClient<>(connectionRequest, Set.of(USER_SPOT_ORDERS_CHANNEL_FORMAT), callback, type, apiClient, objectMapper);
+
+    /**
+     * Subscription on spot (Limit/Market) orders stream for your Account. Stream push all updates to users orders on a symbol
+     */
+    public  WebSocketClient<Order> userSpotTradesSubscription(Request connectionRequest, WebSocketCallback<Order> callback) {
+        return new WebSocketClient<>(connectionRequest, Set.of(USER_SPOT_ORDERS_CHANNEL_FORMAT), callback, Order.class, apiClient, objectMapper);
     }
+    //</editor-fold>
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java:171]: same here

**Review by Wayruha — CHANGES_REQUESTED** [review]

File: src/main/java/trade/wayruha/mexc/constant/ApiEndpoints.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
+package trade.wayruha.mexc.constant;
+
+public class ApiEndpoints {
```
Comment (Wayruha) [inline src/main/java/trade/wayruha/mexc/constant/ApiEndpoints.java:3]: Should we call it something like WSSubscriptions or so?

File: src/main/java/trade/wayruha/mexc/enums/OrderStatus.java (hunk 2 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/enums/OrderStatus.java:11]: Yeap, from WS we get numbers instead of normal string.
 see **trade.wayruha.mexc.dto.Order#setOrderTypeByAlias**

File: src/main/java/trade/wayruha/mexc/dto/Order.java (hunk 1 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/dto/Order.java:20]: It's related, see method  **trade.wayruha.mexc.ws.WSClientFactory#userSpotTradesSubscription**

File: src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java (hunk 4 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java:39]: In order to delimit public and private(personal) method. It's convenient thing in IDE

File: src/main/java/trade/wayruha/mexc/enums/TopLimit.java (hunk 5 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/enums/TopLimit.java:8]: As you wish. It's represent "Top bids and asks, Valid are 5, 10, or 20."

File: src/main/java/trade/wayruha/mexc/constant/ApiEndpoints.java (hunk 7 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/constant/ApiEndpoints.java:3]: I thoght that we could put heree all our API routes and use this constants. 

Like take the path from here 
 @GET("api/v3/exchangeInfo")
 trade.wayruha.mexc.service.endpoint.MarketDataAPI#getExchangeInfo

File: src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java (hunk 3 above)
Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java:142]: Ok. Suggest the solution for filling symbol to inner DataModel from wrapper MexcWSResponse.class

Comment (KhrulSergey) [reply src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java:142]: ** they will add ..subscription**
You are very optimistic)

File: src/main/java/trade/wayruha/mexc/enums/OrderStatus.java (hunk 2 above)
Comment (Wayruha) [reply src/main/java/trade/wayruha/mexc/enums/OrderStatus.java:11]: this dto is (also|primarily) used in REST Get Orders, where they send us a string, e.g., "NEW". Need to make sure OrderStatus we don't break that functionality

File: src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java (hunk 3 above)
Comment (Wayruha) [reply src/main/java/trade/wayruha/mexc/ws/WebSocketClient.java:142]: so far I have a major concern about this...
I believe we should not include a field `symbol` to the base dto as eventually (and undoubtedly) there will be responses which won't have it.

I can't edit the PR, so could you please update it? let's remove any 'bussiness-related' field (`symbol`) from WSBaseDto.
I can also do it by my own but only after mergin the PR and then ammending (to make history looks good) but it does not worth it...

File: src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java (hunk 4 above)
Comment (Wayruha) [reply src/main/java/trade/wayruha/mexc/ws/WSClientFactory.java:39]: which IDE do you use? cause my IDEA does not recognize this comments and just displays it as plain text comment
